src/App.tsx

```diff
@@ -1,24 +1,216 @@
+    setError(errorMessage);
+    setTimeout(() => {
+      setError(null);
+    }, 3000);
+  };
+
+  useEffect(() => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+  }, [loadingTodoIds]);
+
+  useEffect(() => {
+    todosService.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => (handleError('Unable to load todos')));
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    return filteredTodos(todos, status);
+  }, [status, todos]);
+
+  const createTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const trimmedTitle = todoTitle.trim();
+
+    if (!trimmedTitle) {
+      handleError('Title should not be empty');
+    }
+
+    if (trimmedTitle) {
+      const tempTodoId = 0;
+
+      setTempTodo({
+        id: tempTodoId,
+        userId: USER_ID,
+        title: trimmedTitle,
+        completed: false,
+      });
+
+      setLoadingTodoIds(curr => [...curr, tempTodoId]);
+
+      todosService.addTodo({
+        userId: USER_ID, title: trimmedTitle, completed: false,
+      })
+        .then(newTodo => {
+          setTodos(currentTodos => {
+            return [...currentTodos, { ...newTodo }];
+          });
+          setTodoTitle('');
+        })
+        .catch(() => handleError('Unable to add a todo'))
+        .finally(() => {
+          setTempTodo(null);
+          setLoadingTodoIds(curr => curr.filter(id => id !== tempTodoId));
+        });
+    }
+  };
+
+  const deleteTodo = (todoId: number) => {
+    setLoadingTodoIds(curr => [...curr, todoId]);
+
+    todosService.deleteTodo(todoId)
+      .then(() => setTodos(
+        currentTodos => currentTodos.filter(todo => todo.id !== todoId),
+      ))
+      .catch(() => handleError('Unable to delete a todo'))
+      .finally(() => {
+        setLoadingTodoIds(curr => curr.filter(id => id !== todoId));
+      });
+  };
+
+  const clearTodos = () => {
+    todos.map(todo => {
+      if (todo.completed) {
+        todosService.deleteTodo(todo.id)
+          .finally(() => titleField.current?.focus());
+      }
+
+      return 1;
+    });
+
+    setTodos(
+      currentTodos => currentTodos.filter(todo => todo.completed !== true),
+    );
+  };
+
+  const updateTodo = (newTodo: Todo): Promise<void> => {
+    setLoadingTodoIds(curr => [...curr, newTodo.id]);
+
+    return todosService.updateTodo(newTodo)
+      .then((todoFromServer) => setTodos(
+        currentTodos => currentTodos.map((todo) => {
+          const updatedTodo = todoFromServer as Todo;
+
+          if (todo.id === updatedTodo.id) {
+            return updatedTodo;
+          }
+
+          return todo;
+        }),
+      ))
+      .catch(() => handleError('Unable to update a todo'))
+      .finally(() => {
+        setLoadingTodoIds([]);
+      });
+  };
+
+  const updateTodoComplete = (todo: Todo) => {
+    const todoUpdated = { ...todo, completed: !todo.completed };
+
+    updateTodo(todoUpdated);
+  };
+
+  const toggleAllTodos = () => {
```

Comment:
your function just change the completed property of todo, so every completed todo becomes uncompleted and every uncompleted becomes completed (check the demo), fix this, checking if there's some uncompleted todo in todos array, if it is, then make all uncompleted todos completed, in other case make every todo uncompleted
![image 14](https://github.com/mate-academy/react_todo-app-with-api/assets/143831895/22def7b3-c34d-4c46-8b4b-1100019d83ae)